src/nodes/hypertable_modify.c

```diff
@@ -93,6 +95,276 @@ get_chunk_dispatch_states(PlanState *substate)
 	return NIL;
 }
 
+#if PG14_GE
+/*
+ * Helper method which returns true if SEGMENT BY column
+ * is NULLABLE. This method also returns the index of
+ * this NULLABLE SEGMENT BY column in compressed chunk.
+ */
+static bool
+is_segmentby_col_null(HypertableModifyState *context, char *column_name, int *attrno)
+{
+	Chunk *curr_chunk = NULL;
+	Chunk *comp_chunk = NULL;
+	Relation chunk_rel;
+	Relation comp_chunk_rel;
+	TupleDesc tupleDesc;
+	int attno = 0;
+	bool is_null = false;
+
+	chunk_rel = relation_open(context->chunk_rte->relid, AccessShareLock);
+	tupleDesc = RelationGetDescr(chunk_rel);
+	/*
+	 * check if SEGMENT BY column is NULLABLE or not based on uncompressed
+	 * chunk tuple descriptor. This is because compressed chunk column
+	 * attributes doesnot have NULLABLE properties.
```

Comment:
Actually, this is incorrect. You can use a NULLABLE column in SEGMENT BY and it will be NULLABLE in the compressed chunk.

In that case, NullTest won't work correctly:
```
postgres=# SELECT count(*) FROM sample_table WHERE sensor_id IS NOT NULL;
 count 
-------
     7
(1 row)

postgres=# UPDATE sample_table SET temperature = 34.21 WHERE sensor_id IS NOT NULL;
UPDATE 0
postgres=# 


```

Reply:
I mean NULLABLE property in column definition. Please refer to Nullable column in below definitions of compressed and uncompressed chunk.
```
postgres=# \d _timescaledb_internal._hyper_7_17_chunk
   Table "_timescaledb_internal._hyper_7_17_chunk"
  Column   |  Type   | Collation | Nullable | Default
-----------+---------+-----------+----------+---------
 time      | integer |           | not null |
 device_id | integer |           |          |
 val       | integer |           |          |
Indexes:
    "_hyper_7_17_chunk_sample_table_time_idx" btree ("time" DESC)
Check constraints:
    "constraint_9" CHECK ("time" >= 0 AND "time" < 10)
Inherits: sample_table

postgres=# \d _timescaledb_internal.compress_hyper_8_19_chunk
                    Table "_timescaledb_internal.compress_hyper_8_19_chunk"
        Column         |                 Type                  | Collation | Nullable | Default
-----------------------+---------------------------------------+-----------+----------+---------
 time                  | _timescaledb_internal.compressed_data |           |          |
 device_id             | integer                               |           |          |
 val                   | _timescaledb_internal.compressed_data |           |          |
 _ts_meta_count        | integer                               |           |          |
 _ts_meta_sequence_num | integer                               |           |          |
 _ts_meta_min_1        | integer                               |           |          |
 _ts_meta_max_1        | integer                               |           |          |
Indexes:
    "compress_hyper_8_19_chunk__compressed_hypertable_8_device_id__t" btree (device_id, _ts_meta_sequence_num)
Inherits: _timescaledb_internal._compressed_hypertable_8
```